runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PipelineTranslation.java

```diff
@@ -81,8 +96,83 @@ public void visitPrimitiveTransform(Node node) {
             .setComponents(components.toComponents())
             .addAllRootTransformIds(rootIds)
             .build();
+    if (!useDeprecatedViewTransforms) {
+      // TODO(JIRA-5649): Don't even emit these transforms in the generated protos.
+      res = elideDeprecatedViews(res);
+    }
     // Validate that translation didn't produce an invalid pipeline.
     PipelineValidator.validate(res);
     return res;
   }
+
+  private static RunnerApi.Pipeline elideDeprecatedViews(RunnerApi.Pipeline pipeline) {
+    // Record data on CreateView operations.
+    Collection<String> viewTransforms = new HashSet<>();
```

Comment:
Why is this `Collection`? IMHO, `Set` would make the set attributes of the implementation clearer.

Reply:
No good reason. Updating now. (I'm going to ensure the postcommits pass before I push.)